doc(v1.7.0): add clone volume features from longhorn UI (#940)
longhorn/website: content/docs/1.7.0/snapshots-and-backups/csi-volume-clone.md

---

Longhorn supports [CSI volume cloning](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/).

> 

Reviewer: How about changing the layout to

```
## Volume Cloning
### Clone a Volume Using YAML
### Clone a Volume Using the Longhorn UI
### Clone Volumes ...
...

## Volume Creation
```

Author: Want to put UI actions under the same block
```
## Volume Cloning
### Clone a Volume Using YAML
....
### Clone Volume Using the Longhorn UI
#### Clone a volume
#### Clone a Volume Using a Snapshot
#### Clone Multiple Volumes (Bulk Cloning)
...
## Volume Creation
```

> ## Via Kubernetes Yaml

Reviewer 2: ```suggestion
## Clone a Volume Using YAML
```

> In addition to the requirements listed at [CSI volume cloning](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/),
> the `cloned-pvc` must have the same `resources.requests.storage` as the `source-pvc`.

> ## Clone a Volume Using the Longhorn UI

Reviewer 3: Would you like to mention buck cloning?

Author: Sure, let me add another block for bulk volume cloning.

### Clone a volume

> 1. Go to the **Volume** screen.

Reviewer: ```suggestion
1. Go to the **Volume** page.
```

2. Select a volume, and then click **Clone Volume** in the **Operation** menu.
3. (Optional) Configure the settings of the new volume.
4. Click **OK**.

### Clone a Volume Using a Snapshot

> 1. Go to the **Volume** screen.

Reviewer: ```suggestion
1. Go to the **Volume** page.
```

Reviewer 2: In many doc style guides, "page" is used for websites while "screen" is used for program consoles. I use screen when writing about the Longhorn and Harvester UIs.

Microsoft Style Guide:
- [page](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/p/page)
- [screen](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/s/screen)

2. Click the name of the volume that you want to clone.
3. In the **Snapshot and Backups** section of the details screen, identify the snapshot that you want to use and then click **Clone Volume**.
4. (Optional) Configure the settings of the new volume.
5. Click **OK**.

{{< figure src="/img/screenshots/snapshots-and-backups/clone-volume-modal.png" >}}

**Note**:
> - The Longhorn UI pre-fills certain fields and prevents you from modifying the values to ensure that those match the settings of the source volume.
> - Longhorn automatically attaches the new volume, clones the source volume, and then detaches the new volume.

### Create a Volume Using a Specific Data Source

> 1. Go to the **Volume** screen.

Reviewer: ```suggestion
1. Go to the **Volume** page.
```

> In addition to the requirements listed at [CSI volume cloning](https://kubernetes.io/docs/concepts/storage/volume-pvc-datasource/),
> the `cloned-pvc` must have the same `resources.requests.storage` as the `source-pvc`.

> ## Via Longhorn UI

Reviewer 2: ```suggestion
## Clone a Volume Using the Longhorn UI
```

> To clone volume from existing volume
> 1. Go to volume page
> 2. Choose a volume and click **Clone Volume** from operation menu
> 3. You can edit the cloned volume configurations and click Ok.

Reviewer 2: ```suggestion
### Clone a Volume

1. Go to the **Volume** screen.

1. Select a volume, and then click **Clone Volume** in the **Operation** menu.

1. (Optional) Configure the settings of the new volume.

1. Click **OK**.
```

> To clone volume from volume snapshot.
> 1. Go to volume page, click volume name to enter volume detail page
> 2. In **Snapshot and Backups** block, click *Clone Volume* from any snapshot or backup
> 3. You can edit the cloned volume configurations and click Ok.

Reviewer 2: ```suggestion
### Clone a Volume Using a Snapshot

1. Go to the **Volume** screen.

1. Click the name of the volume that you want to clone.

1. In the **Snapshot and Backups** section of the details screen, identify the snapshot that you want to use and then click **Clone Volume**.

1. (Optional) Configure the settings of the new volume.

1. Click **OK**.
```

Reviewer 2: > 2. In **Snapshot and Backups** block, click *Clone Volume* from any snapshot or backup

I don't know this means exactly because the screenshot does not show it. I'm guessing you are referring to a section that lists available snapshots and backups.

Author: yeah, the available action when click snapshot or backup.
<img width="1428" alt="image" src="https://github.com/longhorn/website/assets/5744158/a398d39e-335d-495b-a38f-161c6fecd9dc">

Reviewer 2: Thank you for the screenshot. The suggested wording should work for the UI flow.

{{< figure src="/img/screenshots/snapshots-and-backups/clone-volume-modal.png" >}}

> > Notes.
> > 1. Size, backing image, data source and data engine are prefilled and disabled since these fields should the same as source volume.
> > 2. Longhorn will auto attach the new cloned volume, perform cloning from source volume, and then detach new volume.

Reviewer 2: ```suggestion
> **Note**:
> - The Longhorn UI pre-fills certain fields and prevents you from modifying the values to ensure that those match the settings of the source volume.
> - Longhorn automatically attaches the new volume, clones the source volume, and then detaches the new volume.
```

> To create new volume from existing volume or volume snapshot
> 1. Go to volume page, click **Create Volume** button
> 2. In **data source** field, choose the data source from existing volume or volume snapshot
> 3. Choose the volume name
> 4. Choose the volume snapshot if data source is **Volume Snapshot**
> 

Reviewer 2: ```suggestion
### Create a Volume Using a Specific Data Source

1. Go to the **Volume** screen.

1. Click **Create Volume**.

1. Select the data source (**Volume** or **Volume Snapshot**) that you want to use.

1. If you selected **Volume Snapshot**, select a snapshot.

1. Specify the volume name.

1. Click **OK**.
```

{{< figure src="/img/screenshots/snapshots-and-backups/create-volume-choose-datasource.png" >}}

## History
- [GitHub Issue](https://github.com/longhorn/longhorn/issues/1815)
- [Longhorn Enhancement Proposal](https://github.com/longhorn/longhorn/pull/2864)

> Available since v1.2.0, available from UI since v1.7.0

Reviewer 2: I don't agree with this change. Such notes indicate when features become available *in general*. We do not provide availability dates for specific interfaces.

Author: Sure, will remove.

### Clone Volume Using the Longhorn UI

#### Clone a volume
1. Go to the **Volume** page.
2. Select a volume, and then click **Clone Volume** in the **Operation** menu.
3. (Optional) Configure the settings of the new volume.
4. Click **OK**.

#### Clone a Volume Using a Snapshot
1. Go to the **Volume** page.
2. Click the name of the volume that you want to clone.
3. In the **Snapshot and Backups** section of the details screen, identify the snapshot that you want to use and then click **Clone Volume**.
4. (Optional) Configure the settings of the new volume.
5. Click **OK**.

{{< figure src="/img/screenshots/snapshots-and-backups/clone-volume-modal.png" >}}

#### Clone Multiple Volumes (Bulk Cloning)
1. Go to the **Volume** page.
2. Select the volume you want to clone.
3. Click **Clone Volume** button on top of the table.
4. (Optional) Configure the settings of the new volumes
5. Click **OK**

**Note**:
> - The Longhorn UI pre-fills certain fields and prevents you from modifying the values to ensure that those match the settings of the source volume.
> - Longhorn automatically attaches the new volume, clones the source volume, and then detaches the new volume.

> ## Volume Creation
> 1. Go to the **Volume** page.
> 2. Click **Create Volume**.
> 3. Select the data source (**Volume** or **Volume Snapshot**) that you want to use.
> 4. If you selected **Volume Snapshot**, select a snapshot.
> 5. Specify the volume name.
> 6. Click **OK**.
> 

Reviewer: ```suggestion
## Volume Creation
1. Go to the **Volume** page.
2. Click on **Create Volume**.
3. Select the data source (**Volume** or **Volume Snapshot**) that you want to use.
4. If you select **Volume Snapshot**, choose a snapshot.
5. Specify the volume name.
6. Click **OK**.

```

Author: Updated the fourth point.
